apps/nextjs-website/src/lib/cmsApi.ts

```diff
@@ -48,88 +49,157 @@ const buildEnv = pipe(
 );
 
 export const getHomepageProps = async () => {
-  const strapiHomepage = await fetchHomepage(buildEnv);
-  return makeHomepageProps(strapiHomepage);
+  return withCache(
+    getCacheKey('getHomepageProps'),
+    async () => {
+      const strapiHomepage = await fetchHomepage(buildEnv);
+      return makeHomepageProps(strapiHomepage);
+    },
+    900 // 15 minutes in seconds
```

Comment:
maybe we should convert this into a env var ot at  least a constant in this file
```suggestion
    900 // 15 minutes in seconds
```